### PR TITLE
Update Knative-build-pipeline e2e tests for OCP 4.0

### PR DESCRIPTION
**Pull request by arilivigni**
- Updated ci-operator configs and Dockerfiles
- Added e2e-tests-openshift.sh

### CODE REVIEW

**Comment by bbrowning** [comment]
I didn't catch this with your updated build-pipeline config for openshift/release but https://github.com/openshift/release/blob/b963df45da543e1e8051c35ca12d0e63d974a405/ci-operator/config/openshift/knative-build-pipeline/openshift-knative-build-pipeline-master.yaml#L79-L80 is wrong. That should look more like https://github.com/openshift/release/blob/b963df45da543e1e8051c35ca12d0e63d974a405/ci-operator/config/openshift/knative-serving/openshift-knative-serving-release-0.3.yaml#L22-L23

The big difference being openshift-installer instead of Ansible and AWS instead of GCP.

**Comment by arilivigni** [comment]
@bbrowning will make that change on the release side sorry I missed that piece

File: openshift/e2e-tests-openshift.sh [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+  header "Knative Build Pipeline Installed successfully"
+}
+
+function create_build_pipeline(){
+  resolve_resources config/ build-pipeline-resolved.yaml
+  oc apply -f build-pipeline-resolved.yaml
+}
+
+function resolve_resources(){
```
Comment (mgencur) [inline openshift/e2e-tests-openshift.sh]: @mvinkler in his PR for this branch (but OCP 3.11) had this function different: https://github.com/openshift/knative-build-pipeline/pull/4/commits/33805aa0309f332b8d76212034197b32f611b700
I think there's something missing.

File: openshift/e2e-tests-openshift.sh [hunk 2]
```diff
@@ -0,0 +1,168 @@
+#!/bin/sh
+
+source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/e2e-tests.sh
+
+set -x
+
+readonly API_SERVER=$(oc config view --minify | grep server | awk -F'//' '{print $2}' | awk -F':' '{print $1}')
```
Comment (sthaha) [inline openshift/e2e-tests-openshift.sh]: `oc config view --minify | awk -F: '/server:/ { gsub(" http.://", ""); print $2}'` may be simpler

File: openshift/e2e-tests-openshift.sh [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  local resolved_file_name=$2
+  local registry_prefix="$OPENSHIFT_REGISTRY/$OPENSHIFT_BUILD_PIPELINE_NAMESPACE/stable"
+  > $resolved_file_name
+  for yaml in $(find $dir -name "*.yaml" | grep -vE $IGNORES); do
+    echo "---" >> $resolved_file_name
+    #first prefix all test images with "test-", then replace all image names with proper repository and prefix images with "knative-build-pipeline"
+    sed -e 's%\(.* image: \)\(github.com\)\(.*\/\)\(test\/\)\(.*\)%\1\2 \3\4test-\5%' $yaml | \
+    sed -e 's%\(.* image: \)\(github.com\)\(.*\/\)\(.*\)%\1 '"$registry_prefix"'\:knative-build-pipeline-\4%' | \
+    # process these images separately as they're passed as arguments to other containers
+    sed -e 's%github.com/knative/build-pipeline/cmd/creds-init%'"$registry_prefix"'\:knative-build-pipeline-creds-init%g' | \
```
Comment (vdemeester) [inline openshift/e2e-tests-openshift.sh]: We'll need to add that `entrypoint` image in those `sed`s too

File: openshift/e2e-tests-openshift.sh [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    done
+  if (( ! tests_finished )); then
+    echo "ERROR: tests timed out"
+    return 1
+  fi
+
+  # Check that tests passed.
+  local failed=0
+  echo ">> Checking test results"
+  for expected_status in succeeded failed; do
```
Comment (vdemeester) [inline openshift/e2e-tests-openshift.sh]: this is not the case anymore in pipeline, we don't check `succeeded` of `failed` (see https://github.com/tektoncd/pipeline/blob/master/test/e2e-tests-yaml.sh#L34). All the yaml tests are supposed to succeed :sweat_smile:

**Comment by mgencur** [comment]
Btw, the master branch of knative-serving has a script for generating the manifest for ci-operator, including the image definitions. See https://github.com/openshift/knative-serving/blob/master/Makefile#L29
It would be nice to have it here as well.

**Pull request closed without merging**